build/common/installer/scripts/tomlparser-common-agent-config.rb

```diff
@@ -81,6 +82,14 @@ def populateSettingValuesFromConfigMap(parsedConfig)
       end
     end
 
```

Comment:
as we discussed, read the ENABLE_CUSTOM_METRICS  envvar and if its disabled, return false in this  function https://github.com/microsoft/Docker-Provider/blob/ci_prod/source/plugins/ruby/CustomMetricsUtils.rb#L10 as this being used in input plugins to generate MDM records.